File: BinaryTrees/16/levelOrder.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

from collections import deque

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        q = deque()
        ans = []
        li = None
        i = -1
        q.append((root, 0))
        while len(q):
            node, level = q.popleft()
            if node.left:
                q.append((node.left, level+1))
            if node.right:
                q.append((node.right, level+1))
            if i != level:
                if li: # Be careful here in JS
                    ans.append(li)
                li = []
                li.append(node.val)
                i += 1
            else:
                li.append(node.val)
        ans.append(li)
        return ans

File: BinaryTrees/16/levelOrder.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  if (!root) return []
  let q = []
  const ans = []
  let li = []
  let i = -1
  q.push([root, 0])
  while(q.length){
    [node, level] = q.shift();
    if (node.left) q.push([node.left, level+1])
    if (node.right) q.push([node.right, level+1])
    if (i !== level) {
      if (li.length) ans.push(li)
      li = []
      li.push(node.val)
      i++;
    } else {
      li.push(node.val)
    }
  }
  ans.push(li);
  return ans;
};
